controller/controller.mustache

```diff
@@ -64,13 +65,22 @@ module.exports = {
   create: ({{name}}) => {
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
+    {{name}}.rev = uuid.v4();
     return r.table('{{pluralName}}').insert({{name}}, {returnChanges: true}).run()
     .then(rethinkdb.firstChange);
   },
   update: ({{name}}) => {
+    if (!{{name}}.rev) return Promise.reject(new httpError('BAD_REQUEST', 'No rev provided'));
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
-    return r.table('{{pluralName}}').update({{name}}, {returnChanges: true}).run()
+    return r.table('{{pluralName}}')
+    .get({{name}}.id)
+    .update(
+      rethinkdb.ifRevMatches({{name}}),
+      {returnChanges: true}
+    )
+    .run()
+    .then(rethinkdb.checkRevError)
```

Comment:
I think we can use an insert here, with a conflict function.  Something like:
```
return r.table('{{pluralName}}')
  .insert({{name}}.id, {
    conflict: conflictResolution
  });

function conflictResolution(id, oldDoc, newDoc) {
  if (newDoc.rev != oldDoc) {
    // throw error, I dunno, however rethink throws errors...
    return oldDoc;
  }
  return newDoc;
}
```

This would also mean you could combine `checkRevError` and `ifRevMatches` into one `conflictResolution` (or whatever) function.